R/types-raw.R

```diff
@@ -15,3 +15,19 @@ bytes <- function(x) {
     abort("`x` should be integerish")
   }
 }
+
+#' Coerce to a raw vector.
+#'
+#' This currently only works with strings, and returns its hexadecimal
+#' representation.
+#'
+#' @param x A string.
+#' @return A raw vector of bytes.
+#' @export
+as_bytes <- function(x) {
+  if (is_bytes(x)) {
```

Comment:
do you mean using a literal `switch(typeof(x), ...)`? I'd prefer if switchpatch was taken as a figure of speech ;)

here we also dispatch on the size, so using switch would make the code slightly harder to read I think